src/environmentd/src/lib.rs

```diff
@@ -362,22 +363,68 @@ impl Listeners {
             })
             .transpose()?;
 
+        let mut openable_adapter_storage = catalog_opener(
+            &config.catalog_config,
+            &config.controller,
+            &config.environment_id,
+        )
+        .boxed()
+        .await?;
+
+        // Initialize the system parameter frontend if `launchdarkly_sdk_key` is set.
+        let system_parameter_sync_config = if let Some(ld_sdk_key) = config.launchdarkly_sdk_key {
+            Some(SystemParameterSyncConfig::new(
+                config.environment_id.clone(),
+                &BUILD_INFO,
+                &config.metrics_registry,
+                config.now.clone(),
+                ld_sdk_key,
+                config.launchdarkly_key_map,
+            ))
+        } else {
+            None
+        };
+        let remote_system_parameters = load_remote_system_parameters(
+            &mut openable_adapter_storage,
+            system_parameter_sync_config.clone(),
+        )
+        .await?;
+
+        let catalog_kind_impl_ld = remote_system_parameters
```

Comment:
Can we pull this out into a helper method, plz?

Reply:
Done